File: ios/Classes/Wrapper/Commands/InApp/InAppIsPausedCommand.swift
//
//  Created by Emarsys
//
import EmarsysSDK

public class InAppIsPausedCommand: EmarsysCommandProtocol {
    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        resultCallback(["success": Emarsys.inApp.isPaused()]);
    }
}

File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/inapp/InAppIsPausedCommand.kt
package com.emarsys.emarsys_sdk.command.inapp

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class InAppIsPausedCommand: EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        resultCallback.invoke(Emarsys.inApp.isPaused, null)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}